kubectl-cloudflow/cmd/deploy.go

```diff
@@ -171,21 +165,25 @@ func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 		util.LogAndExit("The application descriptor is invalid, %s", err.Error())
 	}
 
-	createOrUpdateCloudflowApplication(cloudflowApplicationClient, applicationSpec, ownerReference)
+	createOrUpdateCloudflowApplication(cloudflowApplicationClient, applicationSpec)
 
 	// When the CR has been created, create a ownerReference using the uid from the stored CR and
 	// then update secrets and service account with the ownerReference
 	storedCR, err := cloudflowApplicationClient.Get(applicationSpec.AppID)
 	if err != nil {
 		util.LogAndExit("Failed to retrieve the application `%s`, %s", applicationSpec.AppID, err.Error())
 	}
-	ownerReference = storedCR.GenerateOwnerReference()
+	ownerReference := storedCR.GenerateOwnerReference()
+
+	if _, err := createOrUpdateServiceAccount(k8sClient, namespace, serviceAccount, ownerReference); err != nil {
+		util.LogAndExit("%s", err)
+	}
 
 	streamletNameSecretMap = deploy.UpdateSecretsWithOwnerReference(ownerReference, streamletNameSecretMap)
 	createOrUpdateStreamletSecrets(k8sClient, namespace, streamletNameSecretMap)
 
 	serviceAccount.ObjectMeta.OwnerReferences = []metav1.OwnerReference{ownerReference}
```

Comment:
This line of code has been added in `createOrUpdateServiceAccount`. It should be remove from here.

Reply:
Fixed.